src/test/java/org/ibp/api/java/impl/middleware/germplasm/GermplasmAttributeServiceImplTest.java

```diff
@@ -0,0 +1,148 @@
+package org.ibp.api.java.impl.middleware.germplasm;
+
+import org.generationcp.middleware.api.attribute.AttributeDTO;
+import org.generationcp.middleware.domain.germplasm.GermplasmAttributeRequestDto;
+import org.generationcp.middleware.pojos.UDTableType;
+import org.generationcp.middleware.pojos.workbench.WorkbenchUser;
+import org.ibp.api.java.impl.middleware.common.validator.AttributeValidator;
+import org.ibp.api.java.impl.middleware.common.validator.GermplasmValidator;
+import org.ibp.api.java.impl.middleware.common.validator.LocationValidator;
+import org.ibp.api.java.impl.middleware.security.SecurityService;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.validation.BindingResult;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+
+@RunWith(MockitoJUnitRunner.class)
+public class GermplasmAttributeServiceImplTest {
+
+	private static final Integer GID = 1;
+	private static final Integer GERMPLASM_ATTRIBUTE_ID = 1;
+	private static final String GERMPLASM_ATTRIBUTE_TYPE = "PASSPORT";
+
+	@Mock
+	private org.generationcp.middleware.api.germplasm.GermplasmAttributeService germplasmAttributeService;
+
+	@Mock
+	private AttributeValidator attributeValidator;
+
+	@Mock
+	private GermplasmValidator germplasmValidator;
+
+	@Mock
+	private LocationValidator locationValidator;
+
+	@Mock
+	private SecurityService securityService;
+
+	@Captor
+	private ArgumentCaptor<Set<String>> setArgumentCaptor;
+
+	@InjectMocks
+	private GermplasmAttributeServiceImpl germplasmAttributeServiceImpl;
+
+
+	@Test
+	public void testGetGermplasmAttributeDtos() {
```

Comment:
same here, please give a meaningful name to the functions and verify only one condition per test